pkg/splunk/client.go

```diff
@@ -93,6 +92,10 @@ func newSplunkLogger(ctx context.Context, url, token, source, hostname string) *
 		},
 	}
 
+	if maximumSize != 0 {
+		sl.maximumSize = maximumSize
+	}
+
```

Comment:
Since this is not a public API, I am thinking a little bit clunky behavior - maximumSize must be passed, but when 0 is passed use the default value. This works nicely with the public API which has `SplunkConfig` which defaults to zero value if you skip it.